homeassistant/components/tessie/cover.py

```diff
@@ -0,0 +1,107 @@
+"""Cover platform for Tessie integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from tessie_api import (
+    close_charge_port,
+    close_windows,
+    open_unlock_charge_port,
+    vent_windows,
+)
+
+from homeassistant.components.cover import (
+    CoverDeviceClass,
+    CoverEntity,
+    CoverEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import TessieDataUpdateCoordinator
+from .entity import TessieEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Tessie sensor platform from a config entry."""
+    coordinators = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        Entity(coordinator)
+        for Entity in (
+            TessieWindowEntity,
+            TessieChargePortEntity,
+        )
+        for coordinator in coordinators
+    )
+
+
+class TessieWindowEntity(TessieEntity, CoverEntity):
```

Comment:
You can't control individual windows?

Reply:
No, you cant :(
The function call lowers or raises all windows